actor/src/main/scala-2.12/org/apache/pekko/util/ByteString.scala

```diff
@@ -827,6 +836,16 @@ sealed abstract class ByteString extends IndexedSeq[Byte] with IndexedSeqOptimiz
    */
   def toArrayUnsafe(): Array[Byte] = toArray
 
+  /**
+   * Return the bytes in this ByteString as an InputStream.
+   *
+   * @return the bytes in this ByteString accessible as an InputStream
+   * @see [[asByteBuffer]]
+   * @since 1.1.0
+   */
+  def asInputStream: InputStream =
+    new SequenceInputStream(asByteBuffers.map(bb => new ByteBufferBackedInputStream(bb)).iterator.asJavaEnumeration)
```

Comment:
this should be an abstract method, right?
```suggestion
  def asInputStream: InputStream
```

Reply:
Agreed as it forces you to implement it for all subtypes which avoids leaving in a default sub-optimal (in terms of performance) implementation if a subtype happens to miss an implementation

Reply:
removed the default impl